src/preset_cli/auth/token.py

```diff
@@ -12,9 +14,27 @@ class TokenAuth(Auth):  # pylint: disable=too-few-public-methods, abstract-metho
     Auth via a token.
     """
 
-    def __init__(self, token: str):
+    def __init__(self, token: str, baseurl: Optional[URL] = None):
         super().__init__()
         self.token = token
+        self.baseurl = baseurl
+
+    def get_csrf_token(self, jwt: str) -> str:
```

Comment:
It's better to create a new `Auth` class, maybe `SupersetJWTAuth`, since the path `"api/v1/security/csrf_token/"` is very specific to Superset. Otherwise this generic class becomes tightly coupled with Superset and it becomes hard to reuse it for other APIs.

Reply:
ohhhhh that makes sense! Just noticed this class is also used by dbt. Going to make these changes. Thank you!